APNS/APNS/APNSPusher.swift

```diff
@@ -205,3 +206,43 @@ extension APNSPusher: URLSessionDelegate {
         completionHandler(.useCredential, cred)
     }
 }
+
+// MARK: - DevAuthToken
+
+private struct DevAuthToken {
+    private let keyID: String
+    private let teamID: String
+    private let p8Digest: String
+    private let authToken: String
+    private let timestamp = Date()
+    /// 20 min to resolve
+    /// [TooManyProviderTokenUpdates](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns)
+    private let providerTokenTTL: TimeInterval = 60 * 20
+    
+    init?(keyID: String, teamID: String, p8Digest: String) {
+        guard let authToken = try? JWT(keyID: keyID,
+                                   teamID: teamID,
+                                   issueDate: timestamp,
+                                   expireDuration: providerTokenTTL).sign(with: p8Digest) else {
+            return nil
+        }
+        self.authToken = authToken
+        self.keyID = keyID
+        self.teamID = teamID
+        self.p8Digest = p8Digest
+    }
+    
+    var isValid: Bool {
+        Date().timeIntervalSince(timestamp) < providerTokenTTL
+    }
+}
+extension DevAuthToken: Equatable {
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.keyID == rhs.keyID && lhs.teamID == rhs.teamID && lhs.p8Digest == rhs.p8Digest
```

Comment:
no `authToken`?

Reply:
The name of this struct was confusing.

I have a check above, when the user sends a push, to check if the last cached "provider" is the same as the new provider since a user could change their "provider identity" (i.e. team ID, keyID, p8) at anytime before it's considered expired at the 20min mark. I renamed this struct to `APNSProvider` to clarify that this is an Apple Developer account identity instead of its "token". This way, I can do an equality check against the "providers" instead of the providers' tokens.